Sources/Sentry/Profiling/Exception.h

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Specto Inc. All rights reserved.
+
+#pragma once
+
+#include "Log.h"
+
+#include <exception>
+#include <functional>
+#include <string>
+
+namespace specto {
+
+/**
+ * Adds an observer (as a function callback) that is called when the C++ exception
+ * killswitch is flipped.
+ * @param f The function to call when the killswitch is flipped.
+ */
+void addCppExceptionKillswitchObserver(std::function<void(void)> f);
```

Comment:
1. Yes
<img width="743" alt="Screenshot 2022-01-28 at 06 23 37" src="https://user-images.githubusercontent.com/2443292/151491907-d4837e0e-4579-46eb-aa4d-b10086462bf6.png">

2. What do you mean by killswitch the SDK? We have `SentrySDK.close()`. It tries to cleanup as much as possible, but I think you can't undo hooking into signal handlers for example.
3. We use the [C++ exception monitor](https://github.com/getsentry/sentry-cocoa/blob/master/Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_CPPException.cpp) from KSCrash. You can try the [macOS-Swift sample app](https://github.com/getsentry/sentry-cocoa/blob/06f40f9b1c0d9c979f7e0f20ec7563b7f40facaf/Samples/macOS-Swift/macOS-Swift/ViewController.swift#L51-L54) to see how C++ exceptions are handled. Here is an [example event](https://sentry.io/organizations/sentry-sdks/issues/2839944993/events/0ca33cb4d0a74ca29e5eb70cb2a65220/?project=5428557). In case you don't have access try this: https://sentry.io/share/issue/b9b6f49ad6c64d929ce801263b645c5b/. We have a few issues that C++ exceptions are not working properly though see https://github.com/getsentry/sentry-cocoa/issues?q=is%3Aopen+is%3Aissue+label%3A%22Language%3A+C%2B%2B%22. I'm happy to improve that.

Reply:
Thank you for the detailed answers 🙏🏽 I've removed our own C++ exception handling for now to avoid over complicating the situation, we can revisit better support for this as a separate task.